internal/filesystem.go

```diff
@@ -217,16 +217,38 @@ func cleanPidFile(pidFile string) {
 	}
 }
 
+// EnsureDir creates all directories along the path excluding the last element.
 func EnsureDir(path string) error {
 	dir, err := filepath.Abs(filepath.Dir(path))
+	if err != nil {
+		return fmt.Errorf("getting absolute path: %w", err)
+	}
+	if err := ensureDir(dir); err != nil {
+		return fmt.Errorf("ensuring dir: %w", err)
+	}
+	return nil
+}
+
+// EnsureDirAll creates all directories along the path.
+func EnsureDirFull(path string) error {
+	dir, err := filepath.Abs(path)
 	if err != nil {
 		return err
 	}
-	if _, err := os.Stat(dir); os.IsNotExist(err) {
-		err = os.MkdirAll(dir, PermUserRWX)
+	if err := ensureDir(dir); err != nil {
+		return fmt.Errorf("ensuring dir: %w", err)
+	}
+	return nil
+}
+
+func ensureDir(path string) error {
+	if _, err := os.Stat(path); os.IsNotExist(err) {
```

Comment:
do you know why is `os.Stat` needed?
`os.MkdirAll` does internally a `Stat` and if the path exists and it is a folder will return no error.
But in our case if at the path something exists, including a file, then the folders are not created anymore.